test/Tests/src/Network/Http_Spec.enso

```diff
@@ -1,332 +1,205 @@
-                {
-                  "headers": {
-                    "Content-Length": "16",
-                    "Content-Type": "application/json",
-                    "User-Agent": "Java-http-client/17.0.7"
-                  },
-                  "origin": "127.0.0.1",
-                  "url": "",
-                  "args": {},
-                  "data": "{ \\"key\\": \\"val\\" }",
-                  "files": null,
-                  "form": null,
-                  "json": {
-                    "key": "val"
-                  }
-                }
-            json_text = '''
-                { "key": "val" }
-            req = Request.new HTTP_Method.Post url_post
-            req_with_body = req.with_json json_text
-            res = HTTP.new.request req_with_body
-            res.code.should_equal HTTP_Status_Code.ok
-            res.body.decode_as_json.should_equal expected_response
+            response = HTTP.post url_post test_file
+            response.at "data" . should_equal test_file.read_text
+
+        Test.specify "Can perform a Text POST with auto-conversion" <|
+            response = HTTP.post url_post "hello world"
+            response.at "data" . should_equal "hello world"
+            response.at "headers" . at "Content-Length" . should_equal "11"
+            response.at "headers" . at "Content-Type" . should_equal "text/plain"
+
+        Test.specify "Can skip auto-parse" <|
+            response = HTTP.post url_post (Request_Body.Text "hello world") try_auto_parse_response=False
+            response.body.bytes.length . should_equal 383
+
+        Test.specify "Can send a custom header" <|
+            response = HTTP.post url_post (Request_Body.Text "hello world") headers=[Header.new "Custom" "asdf"]
+            response.at "data" . should_equal "hello world"
+            response.at "headers" . at "Content-Length" . should_equal "11"
+            response.at "headers" . at "Content-Type" . should_equal "text/plain"
+            response.at "headers" . at "Custom" . should_equal "asdf"
+
+        Test.specify "Cannot POST when output context is disabled" <|
+            Context.Output.with_disabled <|
+                HTTP.post url_post (Request_Body.Text "hello world") . should_fail_with Forbidden_Operation
+
+        Test.specify "Can perform a Request_Body.Text PUT" <|
+            url_put = base_url_with_slash + "put"
+            response = HTTP.post url_put (Request_Body.Text "hello world") method=HTTP_Method.Put
+            response.at "data" . should_equal "hello world"
+            response.at "headers" . at "Content-Length" . should_equal "11"
+            response.at "headers" . at "Content-Type" . should_equal "text/plain"
+
+        Test.specify "Can perform a Request_Body.Text PATCH" <|
+            url_patch = base_url_with_slash + "patch"
+            response = HTTP.post url_patch (Request_Body.Text "hello world") method=HTTP_Method.Patch headers=[Header.content_type "application/diff"]
```

Comment:
What happens if I do:
```suggestion
        Test.specify "Can perform a Request_Body.Text PATCH" <|
            url_patch = base_url_with_slash + "patch"
            response = HTTP.post url_patch (Request_Body.Text "hello world" content_type="foo") method=HTTP_Method.Patch headers=[Header.content_type "application/diff"]
```
?

Will the content type from `Request_Body` or headers "win"? Or will both headers be included (IIRC HTTP allows for duplicated headers).

IMO it would be good to have a test 'documenting' what is the behaviour here.

Reply:
In the case of httpbin.org, the first content type wins. But it's completely dependent on the server implementation

Reply:
We should merge the headers in the request stage - I think if present in the headers we shouldn't add another one.

Reply:
Multiple headers are explicitly allowed in the spec. Maybe a warning?

Reply:
Multiple headers specified by the user yes - but if the user specifies a content type header, that should take precedence over the content type determined by the library.

Especially as, I think with `Request_Body.Text` there is no way to specify _no_ content type, right?

Reply:
Alternately, we could remove the content_type field from Request_Body.Text (the only constructor that has it), because the user always has the option of passing in the content type via the headers parameter.

I suggest this because the alternative is to replace the request body's content type with an override from the list, also handling cases when multiple content types are specified in the list, and it seems redundant to have both methods of specifying a content type.